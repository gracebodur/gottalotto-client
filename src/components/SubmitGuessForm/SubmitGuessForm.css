form {
    background-color: #003F5A;
}

.submit {
    padding: 16px;
	height: auto;
	margin: 0 auto;
    width: 100%;
}

.typical-submit {
    position: absolute;
    color: #003F5A;
}

.submit h2 {
    padding-top: 35px;
	padding-bottom: 10px;
	font-size: 2em;
	text-align: center;
	letter-spacing: 0.2em;
}

.submitForm {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
    margin-top: 10%;
}

.submitText {
    color: white;
}

.body {
    -webkit-font-kerning: none;
    font-kerning: none;
    font-size: 14px;
    padding: 2em;
    text-align: center;
}

.body * {
    box-sizing: border-box;
    -webkit-box-sizing: border-box;
}

.body h1 {
    font-size: 1.5em;
    font-weight: 500;
    line-height: 1.6;
    margin: 0 0 2em;
}

/* .body input[type="number"] {
    -moz-appearance: textfield;
}

.body input[type="number"]::-webkit-inner-spin-button,
.body input[type="number"]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
} */

.body .pt_Quantity {
    display: inline-block;
    margin: 0 0 10px;
    position: relative;
}

.body .pt_Quantity input {
    color: #443017;
    border-radius: 2px;
    background-color: #ffffff;
    border: solid 1px #b7b7b7;
    font-family: "Montserrat", sans-serif;
    font-size: 15px;
    font-weight: 300;
    min-width: 45px;
    height: 30px;
    float: left;
    display: block;
    padding: 0 15px 0 2px;
    margin: 0;
    text-align: center;
    width: 70px;
}

.body .pt_Quantity input:focus {
    outline: 0;
}

.body .pt_Quantity .pt_QuantityNav {
    float: left;
    position: relative;
    height: 30px;
}

.body .pt_Quantity .pt_QuantityNav .pt_QuantityButton {
    position: relative;
    cursor: pointer;
    border-left: 1px solid #b7b7b7;
    width: 15px;
    text-align: center;
    color: #333;
    font-size: 14px;
    font-family: "Trebuchet MS", Helvetica, sans-serif !important;
    line-height: 1.1;
    -webkit-transform: translateX(-100%);
    transform: translateX(-100%);
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

.body .pt_Quantity .pt_QuantityNav .pt_QuantityButton:active {
    box-shadow: inset 0 0 5px rgba(0, 0, 0, 0.2);
    -webkit-box-shadow: inset 0 0 5px rgba(0, 0, 0, 0.2);
}

.body .pt_Quantity .pt_QuantityNav .pt_QuantityButton.pt_QuantityUp {
    position: absolute;
    height: 50%;
    top: 0;
    border-bottom: 1px solid #b7b7b7;
}

.body .pt_Quantity .pt_QuantityNav .pt_QuantityButton.pt_QuantityDown {
    position: absolute;
    bottom: 0px;
    height: 50%;
}