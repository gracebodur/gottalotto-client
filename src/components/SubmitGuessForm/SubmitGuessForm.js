import React, { Component } from 'react'
import GuessesApiService from '../../services/guesses-api-service'
import UserContext from '../../contexts/userContext'

class SubmitGuessForm extends Component {
	static contextType = UserContext

	state = {
		guess_1: "",
		guess_2: "",
		guess_3: "",
		guess_4: "",
		guess_5: "",
		power_ball: "",
		message: ""
	}

	handleSubmit = (ev, data) => {
		ev.preventDefault();
		GuessesApiService.postGuess(data)
			.then()
			.catch(this.context.setError);
		const { location, history } = this.props;
		const destination = (location.state || {}).from || "/";
		history.push(destination);
	};

	handleChange = event => {
		const { name, value, type, checked } = event.target;
		type === "checkbox"
			? this.setState({ [name]: checked })
			: this.setState({ [name]: value });
	};

	render() {
		const { guess_1, guess_2, guess_3, guess_4, guess_5, power_ball, message } = this.state;
		console.log('context', this.context)

		return (
			<div>
				<form onSubmit={event => {
					const data = this.state;
					this.handleSubmit(event, data);
				}}>
					<legend>Select your 6 numbers</legend>
					<p>{guess_1}, {guess_2}, {guess_3}, {guess_4}, {guess_5}, {power_ball}</p>
					<p>{message}</p>
					<input type="number" value={guess_1} name='guess_1' min='1' max='69' onChange={this.handleChange} required />
					<input type="number" value={guess_2} name='guess_2' min='1' max='69' onChange={this.handleChange} required />
					<input type="number" value={guess_3} name='guess_3' min='1' max='69' onChange={this.handleChange} required />
					<input type="number" value={guess_4} name='guess_4' min='1' max='69' onChange={this.handleChange} required />
					<input type="number" value={guess_5} name='guess_5' min='1' max='69' onChange={this.handleChange} required />
					<input type="number" value={power_ball} name='power_ball' min='1' max='26' onChange={this.handleChange} required />
					<textarea
						type="text"
						value={message}
						name="message"
						maxLength="280"
						placeholder='Enter message here'
						onChange={this.handleChange}
						required />
					<button>Submit Guess!</button>
				</form>
			</div>
		)
	}

}

export default SubmitGuessForm