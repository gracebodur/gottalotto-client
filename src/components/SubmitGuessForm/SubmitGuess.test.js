import React from "react";
import toJson from "enzyme-to-json";
import SubmitGuessForm from "./SubmitGuessForm";

describe(`SubmitGuessForm component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a SubmitGuessForm by default", () => {
		const wrapper = <SubmitGuessForm />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the SubmitGuessForm given props", () => {
		const wrapper = <SubmitGuessForm {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});