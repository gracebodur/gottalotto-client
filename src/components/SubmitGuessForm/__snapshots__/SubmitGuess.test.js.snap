// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitGuessForm component renders a SubmitGuessForm by default 1`] = `undefined`;

exports[`SubmitGuessForm component renders the SubmitGuessForm given props 1`] = `undefined`;
